Documentation/internals/security-identities.rst

```diff
@@ -0,0 +1,62 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+*******************
+Security Identities
+*******************
+
+Security identities are generated from labels. They are stored as ``uint32``,
+which means the maximum limit for a security identity is ``2^32 - 1``. The
+minimum security identity is ``1``.
+
+.. note::
+
+   Identity 0 is not a valid value. If it shows up in Hubble output, this means
+   the identity was not found. In the eBPF datapath, it has a special role
+   where it denotes "any identity", i.e. as a wildcard allow in policy maps.
+
+Security identities span over several ranges, depending on the context:
+
+1) Cluster-local
+2) ClusterMesh
+3) Identities generated from CIDR-based policies
+
+Cluster-local identities (1) range from ``1`` to ``2^16 - 1``. The lowest
+values, from ``1`` to ``255``, correspond to the reserved identity range.  See
+the `internal code documentation
+<https://pkg.go.dev/github.com/cilium/cilium/pkg/identity#NumericIdentity>`__
+for details.
+
+For ClusterMesh (2), 8 bits are used as the ``cluster-id`` which identifies the
+cluster in the ClusterMesh, into the 3rd octet as shown by ``0x00FF0000``. This
+does not apply to CIDR identities however, see (3).
```

Comment:
super minor nit, Another more generic way to phrase the last sentence is to define that ClusterMesh identities must set the uppermost byte to 0. Then it's clear that they are distinct from the local identities below, and if we ever decide to allocate meaning to the other uppermost bits, no meaning would change.

Reply:
Addressed.